pkg/util/provider/machinecontroller/controller.go

```diff
@@ -47,6 +47,9 @@ const (
 	// MCMFinalizerName is the finalizer used to tag dependencies before deletion
 	// of the object. This finalizer is carried over from the MCM
 	MCMFinalizerName = "machine.sapcloud.io/machine-controller-manager"
+	// NodeFinalizer is the finalizer used to tag node dependencies before deletion
+	// Added to resolve https://github.com/gardener/machine-controller-manager/issues/1051
+	NodeFinalizer = "node.machine.sapcloud.io/machine-controller"
```

Comment:
To keep names consistent with other finaliser constants
```suggestion
	NodeFinalizerName = "node.machine.sapcloud.io/machine-controller"
```